text/0000-pack-buildpack-new-alternatives.md

```diff
@@ -0,0 +1,272 @@
+
+`pack buildpack create` supports end-users who wish to scaffold a new buildpack.  The operation of `pack buildpack create` requires Internet access to succeed.  `pack buildpack create --template URL` allows projects (such as Paketo, Heroku and others) to define skeleton projects for new buildpacks.
+
+Replacing `pack buildpack new` with `pack buildpack create` allows buildpack authors to more-easily create new buildpacks in their chosen language and framework.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Project scaffolding is [very](https://cookiecutter.readthedocs.io/) [popular](https://yeoman.io/) [in](https://github.com/kowainik/summoner) [other](https://crates.io/crates/cargo-scaffold) [ecosystems](https://github.com/golang-standards/project-layout).  Scaffolding systems are employed to ease onboarding of new developers.  Within `pack` this feature is targeted at onboarding buildpack authors.
+
+`pack buildpack create` creates the same buildpack scaffolding as `pack buildpack new`.  However, `pack buildpack create --template URL` can be used to access alternative project scaffolding.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The design is modeled on [cookiecutter](https://cookiecutter.readthedocs.io/en/1.7.2/) with reference to [springerle](https://github.com/carlmjohnson/springerle) -- a similar implementation in golang -- and [create-go-app](https://github.com/create-go-app/).  We do not want to `os.Exec` a python subprocess to run cookiecutter as this would require availability of a python runtime environment.  Instead we propose to borrow heavily from create-go-app, and generate the default shell scaffolding from a cloned git repoistory.
+
+The `pack buildpack new` project scaffolding accepts `--api`, `--path` and `--stacks` command line flags and the buildpack id as a positional argument.  These command line flags and positional arguments will be replaced with user prompts `pack buildpack create`.
+
+A full session includes the terminal prompts that the project scaffolding tool asks of the end user:
+
+```command
+$ pack buildpack create
+✔ Enter an ID for this buildpack: example/bash
+✔ Enter a directory in which to scaffold the project: bash_buildpack
+Use the arrow keys to navigate: ↓ ↑ → ←
+? Choose the buildpack API version (use the default if you are unsure):
+  ▸ 0.7
+    0.8
+✔ Enter a stack this buildpack will support by default: io.buildpacks.samples.stacks.bionic
+
+Created project in bash_buildpack
+```
+
+The user can skip prompts by providing `--arg key=value` as command line flags. Each of the prompts from the previous interactive invocation can be skipped by providing appropriate pairs of key and value.
```

Comment:
How will an end user discover these args without understanding the prompt.toml section? Should we ensure that the `--help` flag when prompted with a template prints out the available  args? Or should we display the name alongside the prompt text so that it is evident?

Reply:
I have an implementation of `--help` support in scafall.  After playing around with it a bit, I'm comfortable updating the RFC in line with your suggestion.  Thanks!